src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,30 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+class Deadline extends Task {
+    private final LocalDate deadline;
+
+    Deadline(String description, LocalDate deadline) {
+        super(description);
+        this.deadline = deadline;
+    }
+
+    private Deadline(String description, LocalDate deadline, boolean isDone) {
+        super(description, isDone);
+        this.deadline = deadline;
+    }
+
+    @Override
+    public Deadline complete() {
+        return new Deadline(description, deadline, true);
```

Comment:
Rather than returning a new deadline object, I think that it is better to implement a method that changes the boolean value. (eg. completeTask() {this.isDone = true}) However, I am not sure how much this violates the OOP principles but do consider it.